source/common/http/conn_manager_impl.cc

```diff
@@ -500,6 +522,23 @@ void ConnectionManagerImpl::doConnectionClose(
   }
 }
 
+void ConnectionManagerImpl::createStartDrainTimer(std::chrono::milliseconds drain_delay) {
+  if (!codec_) {
+    stats_.named_.downstream_cx_drain_close_.inc();
+    doConnectionClose(Network::ConnectionCloseType::FlushWrite, absl::nullopt, "");
+    return;
+  }
+
+  start_drain_timer_ = read_callbacks_->connection().dispatcher().createTimer([this]() -> void {
+    if (drain_state_ != DrainState::NotDraining) {
+      return;
+    }
+    stats_.named_.downstream_cx_drain_close_.inc();
```

Comment:
I would move this into the `startDrainSequence()` function and then you consolidate this stat increment.